drivers/mbox/mbox_nrf_bellboard_local.c

```diff
@@ -0,0 +1,194 @@
+	(void)nrfy_bellboard_events_process(bellboard, int_pend);
+
+	for (uint8_t i = 0U; i < NRF_BELLBOARD_EVENTS_TRIGGERED_COUNT; i++) {
+		if ((int_pend & BIT(i)) != 0U) {
+			if (grp->cb[i] != NULL) {
+				grp->cb[i](DEVICE_DT_INST_GET(0), i, grp->user_data[i], NULL);
+			}
+		}
+	}
+}
+
+static inline bool bellboard_local_is_id_valid(uint8_t grp_idx, uint8_t evt)
+{
+	return (grp_idx < BELLBOARD_GROUPS) && (grps[grp_idx] != NULL) &&
+	       (evt < NRF_BELLBOARD_EVENTS_TRIGGERED_COUNT);
+}
+
+static uint32_t bellboard_local_max_channels_get(const struct device *dev)
+{
+	uint32_t max_channels = 0U;
+
+	ARG_UNUSED(dev);
+
+	for (uint8_t i = 0U; i < BELLBOARD_GROUPS; i++) {
+		if (grps[i] != NULL) {
+			max_channels += NRF_BELLBOARD_EVENTS_TRIGGERED_COUNT;
+		}
+	}
+
+	return max_channels;
+}
+
+static int bellboard_local_register_callback(const struct device *dev, uint32_t id,
+					     mbox_callback_t cb, void *user_data)
+{
+	struct mbox_bellboard_local_group *grp;
+	uint8_t grp_idx = MBOX_ID_GRP_GET(id);
+	uint8_t evt = MBOX_ID_EVT_GET(id);
+
+	ARG_UNUSED(dev);
+
+	if (!bellboard_local_is_id_valid(grp_idx, evt)) {
+		return -EINVAL;
+	}
+
+	grp = grps[grp_idx];
+
+	grp->cb[evt] = cb;
+	grp->user_data[evt] = user_data;
+
+	return 0;
+}
+
+static int bellboard_local_set_enabled(const struct device *dev, uint32_t id, bool enable)
+{
+	struct mbox_bellboard_local_group *grp;
+	uint8_t grp_idx = MBOX_ID_GRP_GET(id);
+	uint8_t evt = MBOX_ID_EVT_GET(id);
+
+	ARG_UNUSED(dev);
+
+	if (!bellboard_local_is_id_valid(grp_idx, evt)) {
+		return -EINVAL;
+	}
+
+	grp = grps[grp_idx];
+
+	if (enable) {
+		uint32_t enabled_mask_prev;
+
+		if ((grp->enabled_mask & BIT(evt)) != 0U) {
+			return -EALREADY;
+		}
+
+		enabled_mask_prev = grp->enabled_mask;
+		grp->enabled_mask |= BIT(evt);
+		nrfy_bellboard_int_enable(bellboard, grp_idx, BIT(evt));
+
+		if (enabled_mask_prev == 0U) {
+			irq_enable(irqs[grp_idx]);
+		}
+	} else {
+		if ((grp->enabled_mask & BIT(evt)) == 0U) {
+			return -EALREADY;
+		}
+
+		grp->enabled_mask &= ~BIT(evt);
+		nrfy_bellboard_int_disable(bellboard, grp_idx, BIT(evt));
+
+		if (grp->enabled_mask == 0U) {
+			irq_disable(irqs[grp_idx]);
+		}
+	}
+
+	return 0;
+}
+
+static const struct mbox_driver_api bellboard_local_driver_api = {
+	.max_channels_get = bellboard_local_max_channels_get,
+	.register_callback = bellboard_local_register_callback,
+	.set_enabled = bellboard_local_set_enabled,
+};
+
+#define BELLBOARD_IRQ_CONNECT(grp, grp_idx)                                                        \
+	COND_CODE_1(DT_INST_IRQ_HAS_NAME(0, grp),                                                  \
+		    (static const uint8_t idx_##grp = grp_idx;                                     \
+		     IRQ_CONNECT(DT_INST_IRQ_BY_NAME(0, grp, irq),                                 \
+				 DT_INST_IRQ_BY_NAME(0, grp, priority), bellboard_local_isr,       \
+				 &idx_##grp, 0);),                                                 \
```

Comment:
Why does the second to last param need to be a pointer to a byte in ROM?

Reply:
indeed, just passed number as address.